spec/lib/jobs/redfish-job-spec.js

```diff
@@ -0,0 +1,174 @@
+            redfishJob = new this.Jobclass({}, { graphId: graphId }, uuid.v4());
+            expect(redfishJob.routingKey).to.equal(graphId);
+        });
+        
+        it("should have a _run() method", function() {
+            expect(redfishJob).to.have.property('_run').with.length(0);
+        });
+
+        it("should have a subscribe redfish command method", function() {
+            expect(redfishJob).to.have.property('_subscribeRedfishCommand').with.length(2);
+        });
+
+        it("should listen for redfish command requests", function(done) {
+            var data = {
+                config: { command: 'power' },
+                uri: 'http://testuri',
+                user: 'user',
+                password: 'password',
+                workItemId: 'testworkitemid',
+                node: 'xyz'
+            };
+            redfishJob.initClient = sinon.stub().resolves(redfishApi);
+            redfishJob.collectChassisData = sinon.stub().resolves(chassisData[1].body);
+            redfishJob._publishRedfishCommandResult = sinon.stub();
+            redfishJob._subscribeRedfishCommand = function(routingKey, callback) {
+                testEmitter.on('test-subscribe-redfish-command', function(data) {
+                    callback.call(redfishJob, data);
+                });
+            };
+
+            redfishJob._run()
+            .then(function() {
+                testEmitter.emit('test-subscribe-redfish-command', data);
+                setImmediate(function() {
+                    try {
+                        expect(redfishJob.collectChassisData).to.be.calledOnce;
+                        done();
+                    } catch (e) {
+                        done(e);
+                    }
+                });
+            });
+        });
+        
+        it("should initialize redfish client", function() {
+            return expect(redfishJob.initClient({
+                username:'user', 
+                password:'password',
+                uri: 'http://testapi'
+            })).to.be.fullfilled;
+        });
+        
+        it("should run collectChassis for each command", function() {
+            redfishApi.listChassisAsync = sandbox.stub().resolves(listChassisData);
+            redfishApi.getPowerAsync = sandbox.stub().resolves(chassisData);
+            redfishApi.getThermalAsync = sandbox.stub().resolves(chassisData);
+            _.forEach(['power', 'thermal'], function(command) {
+                return redfishJob.collectChassisData(redfishApi, command)
+                .then(function(data) {
+                    expect(data[0]).to.equal(chassisData[1].body);
+                });
+            });
+        });
+        
+        it("should fail collectChassis for unknown command", function() {
+            redfishApi.listChassisAsync = sandbox.stub().resolves(listChassisData);
+            return redfishJob.collectChassisData(redfishApi, 'unknown')
+            .catch(function(data) {
```

Comment:
Should explicitly fail the test if `redfishJob.collectChassisData` is fulfilled.